Update the docs for the directory structure changes (#813)
containers/youki: docs/src/developer/unwritten_rules.md

Make sure that you comment copiously, and explain the peculiar behavior of your code so that others can understand why certain code is written the way it is. Also make sure to add doc comments and examples for `pub` items in the crates, so that users can find it from the docs generated by `cargo doc`.

#### Scripts

> In any script, any makefile etc, make sure to `set -e` at start of it. This will make script abort after any command fails, rather than continuing with incorrect state and creating knock-on errors.

Reviewer: ```suggestion
In any script, any makefile etc, make sure to `set -e` at the start. This will abort the script after any command fails, rather than continuing with incorrect state and creating knock-on errors.
```
